analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/TestConstraints.cs

```diff
@@ -45,4 +45,32 @@ internal class DummyConstraint : SymbolicConstraint
 
         private DummyConstraint() { }
     }
+
+    internal class PreserveOnFieldResetConstraint : SymbolicConstraint
```

Comment:
This should get simplified a lot once it's a property. Basically down to a plain class with one override

Reply:
It's still too complicated. All we really need is 
```
    internal class PreserveOnFieldResetConstraint : SymbolicConstraint
    {
        public static readonly PreserveOnFieldResetConstraint Default = new();

        public override SymbolicConstraint Opposite => null;
        public override bool PreserveOnFieldReset { get; } = true;
        protected override string Name { get; } = "PreserveOnFieldReset";
    }
```